libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt

```diff
@@ -7,13 +7,16 @@ import com.mapbox.navigation.base.typedef.RoundingIncrement
 import com.mapbox.navigation.base.typedef.TimeFormatType
 
 const val DEFAULT_NAVIGATOR_POLLING_DELAY = 1500L
+const val DEFAULT_FASTER_ROUTE_DETECTOR_INTERVAL = 2 * 60 * 1000L // 2 minutes
 
 class NavigationOptions private constructor(
     @RoundingIncrement private val roundingIncrement: Int,
     @TimeFormatType private val timeFormatType: Int,
     private val navigatorPollingDelay: Long,
     private val distanceFormatter: DistanceFormatter?,
-    private val onboardRouterConfig: MapboxOnboardRouterConfig?
+    private val onboardRouterConfig: MapboxOnboardRouterConfig?,
+    private val enableFasterRouteDetection: Boolean,
```

Comment:
Can we remove this from a big NavigationOptions constructor?

Ideally we should be able to enable or disable faster route detection independently. Customers will want to be able to give their users the ability to enable/disable the feature without requiring the trip to start over.

Reply:
That's a good point. For now after discussing with @abhishek1508 we're going implement 👇 https://github.com/mapbox/mapbox-navigation-android/pull/2462#issuecomment-586020114

> Link faster route timer lifecycle to observer `register` / `unregister`

so it can be removed from here and enabled / disabled as part of `register` and `unregister` APIs. For `fasterRouteDetectorInterval` we can leave it as is (initialized at `MapboxNavigation` construction time and attached to its lifecycle). If we feel like clients want to change the interval dynamically we can think about it later and iterate in a future PR. How does that sound @kmadsen?